### PR TITLE
PUD-816: Enable prometheus metrics and include upstream health metrics

### PR DESCRIPTION
**Pull request by dazoakley**
TODO: Expose the metrics to prometheus once I've confirmed the http request timing is appropriate on the cluster.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/health/BankHolidayHealth.kt [hunk 1]
```diff
@@ -7,5 +7,6 @@ import org.springframework.web.reactive.function.client.WebClient
 @Component("bankHoliday")
 class BankHolidayHealth(
   webClient: WebClient,
+  @Value("bankHoliday") componentName: String,
```
Comment (philsegal) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/health/BankHolidayHealth.kt:10]: Is there a reason to have the component name dynamic here but static in other places? Be nice to be consistent? Its a shame we cant reuse the `@Component` name

Comment (dazoakley) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/health/BankHolidayHealth.kt:10]: Can you use the name from `@Component` - genuine question as that would be much better... 😄 

This is the first bit o' Kotlin/Java I've done in well over a decade so there's lots about stuff I don't know

**Review by philsegal — APPROVED** [review]
Be nice to reuse the ComponentName, but am assuming that wasn't an option.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/health/BankHolidayHealth.kt (hunk 1 above)
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/health/BankHolidayHealth.kt:10]: Lets merge like this, I'll have a look around but initial query suggests its not an option

**Pull request merged**